Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -36,10 +36,13 @@ extension PackageGraph {
 
         // Create a map of the manifests, keyed by their identity.
         let rootManifestsMap = root.manifests
-        let externalManifestsMap = externalManifests.map{ (identityResolver.resolveIdentity(for: $0.packageLocation), $0) }
-        let manifestMap = rootManifestsMap.merging(externalManifestsMap, uniquingKeysWith: { lhs, rhs in
-            return lhs
-        })
+        var manifestMap = rootManifestsMap
+        externalManifests.forEach {
```

Comment:
A nitpick, but would it be simpler to start with the externals and then just merge the roots into it, replacing any entry with the same identity?